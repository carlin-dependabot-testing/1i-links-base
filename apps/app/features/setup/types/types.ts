import { type UseFormReturn } from 'react-hook-form'

import { type InferSettingsSchema } from '@/features/settings/schemas/settings.schema'

export interface StepProps {
  form: UseFormReturn<InferSettingsSchema>
  onNext?: () => void
  onBack?: () => void
  onSkip?: () => void
  isSkipping?: boolean
}

export interface Step {
  title: string
  description: string
}

export const SETUP_STEPS: Step[] = [
  { title: 'Welcome', description: 'Welcome to the setup wizard' },
  { title: 'Basic Settings', description: 'Configure your application basics' },
  { title: 'Request Settings', description: 'Configure link request options' },
  { title: 'Template Setup', description: 'Choose your initial data setup' }
]
