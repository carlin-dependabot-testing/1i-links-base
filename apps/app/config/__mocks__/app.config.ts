const appConfig = {
  bannerText: ''
  // Add other config properties as needed
}

export default appConfig
