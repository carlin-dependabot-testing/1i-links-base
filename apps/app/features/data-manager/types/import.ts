export type ImportResult = {
  success: {
    added: Array<{ name: string; url: string }>
    updated: Array<{ name: string; url: string }>
  }
  failed: Array<{
    name: string
    url: string
    reason: 'duplicate' | 'invalid' | 'error'
  }>
  categories: {
    new: string[]
    existing: string[]
  }
}
