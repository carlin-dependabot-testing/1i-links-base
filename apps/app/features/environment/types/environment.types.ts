import { type z } from 'zod'

import { type environmentEnum } from '@/features/settings/schemas/settings.schema'

// Derive the type from the Zod enum
export type Environment = z.infer<typeof environmentEnum>
