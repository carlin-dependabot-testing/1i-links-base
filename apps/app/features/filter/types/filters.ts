export interface FilterOption {
  id: string
  label: string
  type?: string
  icon?: React.ComponentType<{ className?: string }>
}
