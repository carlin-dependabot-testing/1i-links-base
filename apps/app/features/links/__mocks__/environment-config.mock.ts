import { type EnvironmentConfig } from '@/features/links/types/environment.types'

export const mockEnvironmentConfig: EnvironmentConfig = {
  environmentColors: {
    production: '#0f0',
    staging: '#ff0',
    integration: '#00f'
  },
  environmentLabels: {
    production: 'Production',
    staging: 'Staging',
    integration: 'Integration'
  },
  enabledEnvironments: ['production', 'staging', 'integration'],
  environmentOrder: ['production', 'staging', 'integration']
}
