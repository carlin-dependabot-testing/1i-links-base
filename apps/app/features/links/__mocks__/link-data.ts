import { type LinksApp } from '@/features/links/types/link.types'

export const mockLinkData: Record<string, LinksApp> = {
  basic: {
    name: 'Test App',
    description: 'Test Description',
    category: 'Category 1',
    color: '#000000',
    environments: {
      production: 'https://test.com'
    },
    tags: ['test-tag'],
    createdAt: '2024-03-20',
    timestamp: '2024-03-20'
  },
  withMultipleEnvironments: {
    name: 'Multi Env App',
    description: 'App with multiple environments',
    category: 'Category 2',
    color: '#000000',
    environments: {
      production: 'https://prod.test.com',
      staging: 'https://staging.test.com',
      integration: 'https://dev.test.com'
    },
    tags: ['test-tag'],
    createdAt: '2024-03-20',
    timestamp: '2024-03-20'
  }
} as const
