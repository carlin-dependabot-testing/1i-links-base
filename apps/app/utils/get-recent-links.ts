import { type LinksApp } from '@/features/links/types/link.types'

export const RECENT_LINKS_LIMIT = 5

export function getRecentLinks(links: LinksApp[]) {
  return [...links]
    .sort(
      (a, b) =>
        new Date(b.timestamp).getTime() - new Date(a.timestamp).getTime()
    )
    .slice(0, RECENT_LINKS_LIMIT)
    .map(link => link.name)
}
