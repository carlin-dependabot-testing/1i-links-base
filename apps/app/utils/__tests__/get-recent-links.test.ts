import { mockLinkData } from '@/features/links/__mocks__/link-data'
import { type LinksApp } from '@/features/links/types/link.types'

import { RECENT_LINKS_LIMIT, getRecentLinks } from '../get-recent-links'

describe('getRecentLinks', () => {
  const mockLinks: LinksApp[] = [
    {
      ...mockLinkData.basic,
      name: 'Link 1',
      timestamp: '2024-03-20T10:00:00Z',
      createdAt: '2024-03-20T10:00:00Z'
    },
    {
      ...mockLinkData.basic,
      name: 'Link 2',
      timestamp: '2024-03-21T10:00:00Z',
      createdAt: '2024-03-21T10:00:00Z'
    },
    {
      ...mockLinkData.withMultipleEnvironments,
      name: 'Link 3',
      timestamp: '2024-03-19T10:00:00Z',
      createdAt: '2024-03-19T10:00:00Z'
    },
    {
      ...mockLinkData.basic,
      name: 'Link 4',
      timestamp: '2024-03-22T10:00:00Z',
      createdAt: '2024-03-22T10:00:00Z'
    },
    {
      ...mockLinkData.withMultipleEnvironments,
      name: 'Link 5',
      timestamp: '2024-03-18T10:00:00Z',
      createdAt: '2024-03-18T10:00:00Z'
    },
    {
      ...mockLinkData.basic,
      name: 'Link 6',
      timestamp: '2024-03-17T10:00:00Z',
      createdAt: '2024-03-17T10:00:00Z'
    }
  ]

  it('getRecentLinks should render correctly', () => {
    const result = getRecentLinks(mockLinks)
    expect(result).toMatchSnapshot()
  })

  it('should return the most recent links limited to RECENT_LINKS_LIMIT', () => {
    const result = getRecentLinks(mockLinks)

    expect(result).toHaveLength(RECENT_LINKS_LIMIT)
    expect(result).toEqual(['Link 4', 'Link 2', 'Link 1', 'Link 3', 'Link 5'])
  })

  it('should return all links when there are fewer than the limit', () => {
    const fewerLinks = mockLinks.slice(0, 3)
    const result = getRecentLinks(fewerLinks)

    expect(result).toHaveLength(3)
    expect(result).toEqual(['Link 2', 'Link 1', 'Link 3'])
  })

  it('should handle empty array', () => {
    const result = getRecentLinks([])

    expect(result).toHaveLength(0)
    expect(result).toEqual([])
  })
})
