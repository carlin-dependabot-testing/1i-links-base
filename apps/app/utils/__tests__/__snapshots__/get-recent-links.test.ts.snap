// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRecentLinks getRecentLinks should render correctly 1`] = `
[
  "Link 4",
  "Link 2",
  "Link 1",
  "Link 3",
  "Link 5",
]
`;
